### ISSUE TITLE
Misuses of cryptographic APIs

### ISSUE THREAD

**Issue by mhp0rtal** [payara/Payara#4070]
Hi
The following lines have cryptographic API misuses. <br> File name => nucleus/common/common-util/src/main/java/org/glassfish/security/common/SSHA.java: Line number => 98: API name => MessageDigest:First parameter (with value "SHA-256") should be any of {SHA-256, SHA-384, SHA-512}<BR>

**Comment by khasunuma** [comment]
This is separated with #4071 and #4072 as different point.

**Comment by mhp0rtal** [comment]
Hi,
Thanks for your attention.
I reported the cryptographic misuses due to my recent research on how developers employ Java Cryptography Architecture (JCA) APIs in Java applications. I utilized some static analysis tools intended for checking cryptographic practices. The aforementioned misuses in those files were obtained from the results of the tools. The severity of the bug is relative. For instance, Even though MD5 is listed on Java documentation, MD5 is not recommended to use in any language anymore as it is considered insecure. An attacker can try billions of candidate passwords per second on a single GPU. Nevertheless, depending on the scenario the same hash algorithm may not be a problem for file integrity purpose.
There are several instances of cryptographic misuses which could lead to a security vulnerability:
*Do not use ECB mode for encryption.
*Do not use a non-random IV for CBC encryption
*Do not use constant encryption keys.
*Do not use constant salts for PBE.
*Do not use fewer than 1,000 iterations for PBE.
*Do not use static seeds to seed SecureRandom(·)
